src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -18,6 +18,10 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
     /// </summary>
     public class ComplexTypeModelBinder : IModelBinder
     {
+        internal const int PropertyDataNotAvailable = 0;
+        internal const int PropertyDataMayBeAvailable = 1;
+        internal const int PropertyDataAvailable = 2;
+
```

Comment:
I'd suggest some different names or **extensive** comments about this means and why it's important

`NoPropertiesCanBind`
`GreedyPropertiesCanBind`
`ValueProviderPropertiesCanBind`

I don't feel super strongly about the specific names, but you get the idea.